src/screens/nano-contract/NanoContractDetail.js

```diff
@@ -116,8 +121,55 @@ function NanoContractDetail() {
     });
   }
 
-  const loadData = async () => {
+  /**
+   * Validates if the transaction is a valid nano contract creation tx
+   * and if it's confirmed by a block already.
+   *
+   * If everything is ok, calls loadData, otherwise calls itself again
+   * in 5s if it's waiting for a block confirmation
+   *
+   * We need this because the state and history are only loaded when the tx is confirmed
+   */
+  const validateAndLoad = async () => {
     setLoading(true);
+    let response;
+    try {
+      response = await wallet.getFullTxById(ncId);
+    } catch (e) {
+      // invalid tx
+      setErrorMessage(t`Transaction is invalid.`);
+      setLoading(false);
+      return;
+    }
+
+    const isVoided = response.meta.voided_by.length > 0;
+    if (isVoided) {
+      setErrorMessage(t`Transaction is voided.`);
+      setLoading(false);
+      return;
+    }
+
+    const isNanoContractCreate = nanoUtils.isNanoContractCreate(response.tx);
+    if (!isNanoContractCreate) {
+      setErrorMessage(t`Transaction must be a nano contract creation.`);
+      setLoading(false);
+      return;
+    }
+
+    const isConfirmed = response.meta.first_block !== null;
+    if (!isConfirmed) {
+      // Wait for transaction to be confirmed
+      setWaitingConfirmation(true);
+      setTimeout(validateAndLoad, 5000);
```

Comment:
I think you should cancel this timeout when the screen is unmounted

In the current implementation it might run forever in the background until the tx is confirmed

Reply:
Thanks, it was leaking. I removed the `setTimeout` with the latest refactor. Can you see it again, please?